pkg/ccl/backupccl/backup_test.go

```diff
@@ -3616,6 +3616,174 @@ func TestBackupRestoreSequence(t *testing.T) {
 	})
 }
 
+func TestBackupRestoreSequenceOwnership(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	const numAccounts = 1
+	_, _, origDB, dir, cleanupFn := BackupRestoreTestSetup(t, singleNode, numAccounts, InitNone)
+	defer cleanupFn()
+	args := base.TestServerArgs{ExternalIODir: dir}
+
+	backupLoc := LocalFoo + `/d`
+	backupLocD2 := LocalFoo + `/d2`
+	backupLocD3 := LocalFoo + `/d3`
+	// Setup for sequence ownership backup/restore tests in the same database.
+	origDB.Exec(t, `CREATE DATABASE d`)
+	origDB.Exec(t, `CREATE TABLE d.t(a int)`)
+	origDB.Exec(t, `CREATE SEQUENCE d.seq OWNED BY d.t.a`)
+	origDB.Exec(t, `BACKUP DATABASE d TO $1`, backupLoc)
+
+	// Setup for cross-database ownership backup-restore tests.
+	origDB.Exec(t, `CREATE DATABASE d2`)
+	origDB.Exec(t, `CREATE TABLE d2.t(a int)`)
+
+	origDB.Exec(t, `CREATE DATABASE d3`)
+	origDB.Exec(t, `CREATE TABLE d3.t(a int)`)
+
+	origDB.Exec(t, `CREATE SEQUENCE d2.seq OWNED BY d3.t.a`)
+
+	origDB.Exec(t, `CREATE SEQUENCE d3.seq OWNED BY d2.t.a`)
+	origDB.Exec(t, `CREATE SEQUENCE d3.seq2 OWNED BY d3.t.a`)
+
+	origDB.Exec(t, `BACKUP DATABASE d2 TO $1`, backupLocD2)
+	origDB.Exec(t, `BACKUP DATABASE d3 TO $1`, backupLocD3)
+
+	t.Run("test sequence owned by table in the same database", func(t *testing.T) {
```

Comment:
Overall, I'd like to see a few more comments on these tests -- like what are we trying to do, what should we see etc.